src/operator/tensor/matrix_op-inl.h

```diff
@@ -376,8 +399,19 @@ void DotBackward_(const nnvm::NodeAttrs& attrs,
     mshadow::Tensor<xpu, 2, real_t> mrhs_data = inputs[2].get<xpu, 2, real_t>(s);
     mshadow::Tensor<xpu, 2, real_t> mlhs_grad = outputs[0].get<xpu, 2, real_t>(s);
     mshadow::Tensor<xpu, 2, real_t> mrhs_grad = outputs[1].get<xpu, 2, real_t>(s);
-    ASSIGN_DISPATCH(mrhs_grad, req[1], dot(mlhs_data.T(), mout_grad));
-    ASSIGN_DISPATCH(mlhs_grad, req[0], dot(mout_grad, mrhs_data.T()));
+    if (param.transpose_a && param.transpose_b) {
```

Comment:
Hmm... sorry I think I made some mistakes. We know that:
```
z = dot(x, y)
dx = dot(dz, y.T)
dy = dot(x.T, dz)
```
Then, we should have:
```
z1 = dot(x.T, y)
dx = dot(dz1, y).T
dy = dot(x, dz1)

z2 = dot(x, y.T)
dx = dot(dz2, y)
dy = dot(x.T, dz2).T

z3 = dot(x.T, y.T)
dx = dot(dz3, y).T
dy = dot(x, dz3).T
```
Basically, I miss the T here. Does  this make sense to you?

Question: how to do `dot(x, y).T` in one mshadow call? Is it equal to one blas call or one blas call + one physical transpose?